fiftyone/core/collections.py

```diff
@@ -9155,6 +9156,35 @@ def values(
         Returns:
             the list of values
         """
+
+        # If we do not need to follow insertion order, we can potentially use a
+        # covered index query to get the values directly from the index and
+        # avoid a COLLSCAN.
+        if not _enforce_natural_order:
+            field = None
+            if isinstance(field_or_expr, str):
+                field = field_or_expr
+            elif isinstance(field_or_expr, list) and len(field_or_expr) == 1:
+                field = field_or_expr[0]
+
+            # Checking _is_full_collection last as it could potentially be expensive.
```

Comment:
`_is_full_collection()` will never be expensive! It exists precisely so that there is a guaranteed fast check to see if we're operating on a full (mongo) collection. Please remove this expense claim 😄

Reply:
oh then i was misremembering 
removed